.github/workflows/rust.yml

```diff
@@ -0,0 +1,81 @@
+---
+name: Rust
+on: [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+  # Minimum supported Rust version (MSRV)
+  ACTION_MSRV_TOOLCHAIN: 1.40.0
+  # Pinned toolchain for linting
+  ACTION_LINTS_TOOLCHAIN: 1.47.0
+
+jobs:
+  tests-stable:
+    name: "Tests, stable toolchain"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: "stable"
+          default: true
+      - name: cargo build
+        run: cargo build
+      - name: cargo test
+        run: cargo test
+      - name: cargo test (failpoints)
+        run: cargo test --features failpoints
+      - name: cargo build (release)
+        run: cargo build --release
+  tests-minimum-toolchain:
+    name: "Tests, minimum supported toolchain (MSRV)"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env['ACTION_MSRV_TOOLCHAIN']  }}
+          default: true
+      - name: cargo test (release)
+        run: cargo test --release
+  linting:
+    name: "Lints, pinned toolchain"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env['ACTION_LINTS_TOOLCHAIN']  }}
+          default: true
+          components: rustfmt, clippy
+      - name: cargo clippy (warnings)
+        run: cargo clippy -- -D warnings
+      - name: cargo fmt (check)
+        run: cargo fmt -- --check -l
+  tests-other-channels:
+    name: "Tests, unstable toolchain"
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        channel:
+          - "beta"
+          - "nightly"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.channel }}
+          default: true
+      - name: cargo build
+        run: cargo build
+      - name: cargo test
+        run: cargo test
```

Comment:
I noticed we split out the stable one out of the matrix and added additional steps there. Are you concerned that those extra steps might fail on beta/nightly? Would be nice if we could combine them.

Reply:
I marked beta+nightly as fallible, but I had to split them because `continue-on-error` is a per-job boolean now. As these can fail anyway, I don't think it's worth doing the extended set of checks we have on stable.

Reply:
Ahh gotcha, missed the `continue-on-error` difference.